### PR TITLE
Ms join org style

### LINKED ISSUE #4784: Apply style for join organization modal
Depends on #4759

### PR DESCRIPTION
**Pull request by fabienSvtr**
## Describe your changes
- Adapt UI depending on mockups
- Create a stepper component for following modal
- Fixed of join organization tests with the new UI

closes #4784

### CODE REVIEW

File: client/src/components/HomePagePopover.vue [hunk 1]
```diff
@@ -114,6 +87,29 @@ async function canDismissModal(data?: any, modalRole?: string): Promise<boolean>
   const { role } = await alert.onDidDismiss();
   return role === ModalResultCode.Confirm;
 }
+
+async function openJoinByLinkModal(): Promise<void> {
+  // const linkModal = await modalController.create({
```
Comment (Max-7) [inline client/src/components/HomePagePopover.vue:92]: Remove comments.

File: client/src/components/JoinOrganization/FinishProcess.vue [hunk 2]
```diff
@@ -1,33 +1,29 @@
...
+    />
+  </div>
 </template>
 
 <script setup lang="ts">
 import {
   caretForward
 } from 'ionicons/icons';
-import { IonList, IonItem, IonIcon } from '@ionic/vue';
+import InformativeText from '../InformativeText.vue';
```
Comment (Max-7) [inline client/src/components/JoinOrganization/FinishProcess.vue:20]: Imports should be absolute (ie `from '@/components/InformativeText.vue'`)

File: client/src/components/WizardStepper.vue [hunk 3]
```diff
@@ -1,35 +1,75 @@
...
+        {{ title }}
+      </ion-text>
+    </div>
   </div>
 </template>
 
 <script setup lang="ts">
-import { IonChip } from '@ionic/vue';
+import { IonText } from '@ionic/vue';
+import WizardStepperStep, { StepStatus } from './WizardStepperStep.vue';
```
Comment (Max-7) [inline client/src/components/WizardStepper.vue:25]: Imports should be absolute.

File: client/src/components/WizardStepperStep.vue [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+<script setup lang="ts">
+import { IonIcon } from '@ionic/vue';
+import { checkmark } from 'ionicons/icons';
+import { defineProps } from 'vue';
+
+defineProps<{
+  status: StepStatus,
+}>();
+
+function getClass(status: string): any {
```
Comment (Max-7) [inline client/src/components/WizardStepperStep.vue:55]: `status` is of type `StepStatus`, not string.

Returned value is of type `string`, not any (probably need to add a default `return '';` at the end of the function.

File: client/src/views/JoinOrganizationModal.vue [hunk 5]
```diff
@@ -189,30 +193,33 @@ import { AvailableDevice } from '@/plugins/libparsec/definitions';
...
-const pageStep = ref(JoinOrganizationStep.WaitForHost);
+const pageStep = ref(JoinOrganizationStep.JoinLink);
+const joinPage = ref();
 const waitPage = ref();
 const hostCodePage = ref();
 const guestCodePage = ref();
 const userInfoPage = ref();
 const passwordPage = ref();
 
+console.log(pageStep.value);
```
Comment (Max-7) [inline client/src/views/JoinOrganizationModal.vue:218]: Remove log.

File: client/src/components/HomePagePopover.vue (hunk 1 above)
Comment (fabienSvtr) [reply client/src/components/HomePagePopover.vue:92]: Done

File: client/src/components/JoinOrganization/FinishProcess.vue (hunk 2 above)
Comment (fabienSvtr) [reply client/src/components/JoinOrganization/FinishProcess.vue:20]: Done

File: client/src/components/JoinOrganization/WaitForHostPage.vue [hunk 6]
```diff
@@ -1,37 +1,33 @@
...
+    />
+  </div>
 </template>
 
 <script setup lang="ts">
 import {
   caretForward
 } from 'ionicons/icons';
-import { IonList, IonItem, IonIcon } from '@ionic/vue';
+import InformativeText from '../InformativeText.vue';
```
[the remark this answers is not in the document]
Comment (fabienSvtr) [reply client/src/components/JoinOrganization/WaitForHostPage.vue:20]: Done

File: client/src/views/JoinOrganizationModal.vue (hunk 5 above)
Comment (fabienSvtr) [reply client/src/views/JoinOrganizationModal.vue:218]: Done

**Pull request merged**